ActivityLog: Filter bar visual tweaks and fixed tablet layout

Initsogar opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Solves https://github.com/Automattic/jetpack-backup-team/issues/259 and https://github.com/Automattic/jetpack-backup-team/issues/260

## Proposed Changes

* Fix layout weirdness in tablets
* Fix Activity Type vertical overflow
* Remove ellipsis in the clear buttons in some resolutions
* Clear buttons now have a border-left to avoid having two icons in close proximity

![CleanShot 2023-11-29 at 21 17 11](https://github.com/Automattic/wp-calypso/assets/1488641/39e66b6f-a792-40d8-987d-b3f810fe7236)


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Spin up a Calypso and Jetpack Cloud live branch
* Navigate to the Activity Log.
* Play around with the filters and ensure it is not breaking UI when date range and activity types.
* To test the changes, I recommend you to play around with different resolutions (using Chrome Developer Tools).

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Initsogar committed:
Fix filterbar style breaking on tablets

Initsogar committed:
Fix activity type button truncate

Initsogar committed:
Add a max width to activity type button span

Initsogar committed:
Separate close buttons from the filter selection

Initsogar commented on client/my-sites/activity/filterbar/style.scss:
```diff
@@ -392,7 +393,7 @@
 		}
 
-		@include break-xlarge {
+		@include breakpoint-deprecated( ">660px" ) {
```
Using this deprecated breakpoint to ensure everything tweaks at the same viewport.

keoshi reviewed (approved):
Changes look great, @Initsogar. Thanks again for refining the UI so quickly.

Only noticed two things we could tweak, but they are non blockers:

1. In some resolutions, the entire search/filtering UI gets pushed to the right by 16px. It seems to come from `.filterbar` styles.

<img width="355" alt="image" src="https://github.com/Automattic/wp-calypso/assets/390760/5b5d612f-b3f7-4344-8c9e-c5c407d32563">


2. The left border on the clear button seems a bit harsh. Changing it from `--color-text-inverted` to `--color-border` leads to a more natural look. Here's a quick before/after

Before | After
--|--
![image](https://github.com/Automattic/wp-calypso/assets/390760/8f94285d-ad7a-4757-a5a9-af48f9478ec8) | ![image](https://github.com/Automattic/wp-calypso/assets/390760/75581056-5f3d-4da1-b793-cca2c814d3d7)

Initsogar committed:
Replace border with color-border on clear button

Initsogar committed:
Remove 16px margin on filterbar, default to 0

Initsogar commented:
Thanks @keoshi! Good catches there.

I just pushed the two fixes:
* Removed the 16px left/right margin on the filterbar
* Switched `--color-text-inverted` to `--color-border`

keoshi commented:
@Initsogar The border looks great and I don't see the extra spacing anymore. However, I think we need to keep the `16px` side margin on screens smaller than 600px; see this:

<img width="540" alt="image" src="https://github.com/Automattic/wp-calypso/assets/390760/a04eb1d8-a771-468d-8db1-8b6ed515b9b8">

Initsogar committed:
Add 16px margin on smaller screens

Initsogar commented:
> @Initsogar The border looks great and I don't see the extra spacing anymore. However, I think we need to keep the `16px` side margin on screens smaller than 600px

My eyes tricked me when I tested it. You are right. Just added the 660px breakpoint (the same we use for switching the filterbar layout into mobile). Fixed!

Initsogar committed:
Merge remote-tracking branch 'origin/trunk' into fix/al-filterbar-styles

Initsogar committed:
Merge remote-tracking branch 'origin/trunk' into fix/al-filterbar-styles
